main-service/src/main/java/ru/practicum/service/CategoryService.java

```diff
@@ -0,0 +1,59 @@
+package ru.practicum.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ru.practicum.dto.category.CategoryDto;
+import ru.practicum.dto.mapper.CategoryMapper;
+import ru.practicum.exception.NotFoundException;
+import ru.practicum.repository.CategoryRepository;
+import ru.practicum.utils.PaginationServiceClass;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryService {
```

Comment:
На классы сервисов советовал бы навесить @Transactional(readOnly = true). Позволяет спрингу делать оптимизацию в его транзакцилнных механизмах.
Вот как авторы спринга комментируют
Reading methods like findAll() and findOne(…) are using @Transactional(readOnly = true) which is not strictly necessary but triggers a few optimizations in the transaction infrastructure (setting the FlushMode to MANUAL to let persistence providers potentially skip dirty checks when closing the EntityManager). Beyond that the flag is set on the JDBC Connection as well which causes further optimizations on that level.

Depending on what database you use it can omit table locks or even reject write operations you might trigger accidentally. Thus we recommend using @Transactional(readOnly = true) for query methods as well which you can easily achieve adding that annotation to you repository interface. Make sure you add a plain @Transactional to the manipulating methods you might have declared or re-decorated in that interface.